12/README.md

```diff
@@ -0,0 +1,165 @@
+
+```ABNF
+EVENT_TYPE = "QUALITY_EVENT" | "INTERFACE_CHANGE" | "PIPE" | "ATTR_CONF_EVENT" |  "CHANGE_EVENT" | "PERIODIC_EVENT" | "ARCHIVE_EVENT" | "USER_EVENT" | "HEARTBEAT"
+
+HEARTBEAT_ENDPOINT = "(upstream server FQDN):port"  
+
+EVENT_ENDPOINT = "(upstream server FQDN):port"
+
+HEARTBEAT_CHANNEL = "(upstream admin device FQDN)/heartbeat" 
+
+EVENT_CHANNEL = "(upstream admin device FQDN)/(attr).EVENT_TYPE[_idl5]" 
+```
+
+## Runtime requirements
+
+Client and server are up and running. Server is reachable from client i.e. may communicate using Request-Reply protocol [RFC-10]. 
+
+## Publisher-Subscriber protocol
+
+![Subscription_1.png](images/Subscription_1.png)
+
+The main goal of the this protocol implementation is to allow client receive events from server:
+
+* Server MUST publish events of EVENT_TYPE
+
+* Server library MUST provide an API to publish events of EVENT_TYPE
+
+* Client SHOULD process events
+
+* Client library SHOULD provide user an API to register user defined event callbacks  
+
+### Negotiation
+
+![Subscription_2.png](images/Subscription_2.png)
+
+Negotiation phase implies that client executes admin's EventSubscriptionChange command:
+
+* Admin MUST implement EventSubscriptionChange command: 
+
+```ABNF
+EventSubscriptionChange = Tango Command
+EventSubscriptionChange_ARGIN-type  = DevVarStringArr 
+EventSubscriptionChange_ARGIN-value = "[ (upstream device), (attribute), "subscribe", EVENT_TYPE]"
+EventSubscriptionChange_ARGOUT-type = DevLong
+EventSubscriptionChange_ARGOUT-value = "(server library release ver)"
+```
+
+**NOTE**: Tango Server Library v9.3.3 force client to explicitly call ZmqEventSubscriptionChange to use ZMQ based implementation [RFC-13]: 
+ 
+```ABNF
+ZmqEventSubscriptionChange = Tango Command
+ZmqEventSubscriptionChangeArgument-type  = DevVarStringArr 
+ZmqEventSubscriptionChangeArgument-value = "[ upstream device, attribute, action ("subscribe|unsubscribe"), EVENT_TYPE]"
+ZmqEventSubscriptionChangeOutput-type = DevVarLongStringArr
+ZmqEventSubscriptionChangeOutput-value = "[(server library release ver), (upstream device idl ver), (zmq_sub_event_hwm), (rate), (ivl), (zmq release ver); (heartbeat_endpoint), (event_endpoint), n*(alternate_heartbeat_endpoint), n*(alternate_event_endpoint),(heartbeat_channel), (event_channel)]"
+```
+
+* Client MUST execute admin's EventSubscriptionChange command to establish Publisher-Subscriber relationship
+
+* Client SHOULD establish tcp? connection to heartbeat/event endpoints
+
+* Client MAY iterate over alternative heartbeat/event endpoints to establish connection if above has failed
+
+* Client MAY use provided heartbeat/event channels to identify event source 
+
+### Subscription 
+
+* Server MUST send events via corresponding EVENT_CHANNEL
+
+* Server MUST send HEARTBEAT event every 9 seconds
+
+* Client MUST execute EventConfirmSubscription command every 600 seconds to indicate its interest in subscription
```

Comment:
In practice, the C++ client library is sending this command every 200 seconds because what is important is for the upstream server to receive it **before** 600 seconds.

Reply:
@bourtemb So it is RECOMMENDED to send this confirmation every 200 s